infra/README.md

```diff
@@ -0,0 +1,130 @@
+  gcloud container clusters get-credentials vdc
+  ```
+
+- Go to the Google Cloud console, VPC networks > internal > Private
+  service connection > Private connections to services, and enable
+  Export custom routes to both connections.
+
+You can now install Hail:
+
+- Create a VM on the internal network, 100GB, Ubuntu 20.04 TLS, allow
+  full access to all Cloud APIs.  10GB will run out of space.  We
+  assume the rest of the commands are run on the VM.
+
+- Standardize file permissions.  This is for docker, which considers
+  permissions for caching.  Run `echo 'umask 022' > ~/.profile`.  You
+  will need to log out/in or run `umask 022`.
+
+- Update $HAIL/config.mk with your infrastructure settings.  You can
+  get settings from the default/global-config secret:
+
+  ```
+  kubectl get secret global-config -o json | jq '.data | map_values(@base64d)'
+  ```
+
+- Install some dependencies on the VM:
+
+  ```
+  sudo apt update
+  sudo apt install -y docker.io python3-pip
+  sudo usermod -a -G docker $USER
+  gcloud -q auth configure-docker
+  gcloud container clusters get-credentials --zone us-central1-a vdc
+  git clone https://github.com/cseed/hail.git
+  python3 -m pip install -r $HOME/hail/docker/requirements.txt
+  ```
+
+  You will have to log out/in for usermod to take effect.
+
+- Run `PROJECT=<gc-project-id>
+  $HAIL/docker/third-party/copy_images.sh`.  This copies some base
+  images from Dockerhub (which now has rate limits) to GCR.
+
+- Generate TLS certificates.  See ../dev-docs/tls-cookbook.md.
+
+- Run `kubectl -n default apply -f $HAIL/ci/bootstrap.yaml`.
+
+- Build the CI utils image.  Run `make push-ci-utils` in $HAIL/ci.
+
+- Deploy the bootstrap gateway.  Run `make deploy` in
+  $HAIL/bootstrap-gateway.
+
+- Create Let's Encrypt certs. Run `make run` in $HAIL/letsencrypt.
+
+- Deploy the gateway.  Run `make deploy` in $HAIL/gateway.
+
+- Deploy the internal-gateway.  Run `make deploy` in $HAIL/internal-gateway.
+
+- Go to the Google Cloud console, API & Services, Credentials.
+  Configure the consent screen.  Add the scope:
+  https://www.googleapis.com/auth/userinfo.email.  Create an OAuth
+  client ID.  Authorize the redirect URI:
+  https://auth.<domain>/oauth2callback.  Download the client secret
+  as client_secret.json.  Create the auth-oauth2-client-secret secret
+  with:
+
+  ```
+  kubectl -n default create secret generic auth-oauth2-client-secret --from-file=./client_secret.json
+  ```
+
+- Bootstrap the cluster by running:
+
+  ```
+  HAIL_SHA=$(git rev-parse HEAD) \
+  HAIL_CI_UTILS_IMAGE=gcr.io/<gcp-project>/ci-utils:latest \
+  HAIL_CI_BUCKET_NAME=dummy \
+  KUBERNETES_SERVER_URL='<k8s-server-url>' \
+  HAIL_DEFAULT_NAMESPACE='default' \
+  HAIL_DOMAIN=<domain> \
+  HAIL_GCP_ZONE=<gcp-zone> \
+  GCP_PROJECT=<gcp-project> \
+  PYTHONPATH=$HOME/hail/ci:$HOME/hail/batch:$HOME/hail/hail/python \
+  python3 $HAIL/ci/bootstrap.py
+  ```
+
+- You may want to add a suitable ssh forward rule to the internal network.
```

Comment:
This should probably precede the previous point, as otherwise one won't be able to ssh into the VM to run the bootstrap command.

How about this?

Add an "allow-ssh" network tag to your VM and then create the following firewall rule to allow connections to your instance in the internal network:
```
gcloud compute firewall-rules create internal-allow-ssh --network internal --action=allow --target-tags=allow-ssh --rules=tcp:22
```

Reply:
OK, I added advice about ssh to the instructions for spinning up the VM.  We have custom firewall rules for ssh that aren't going to generalize and I think different users will have different security postures, so OK with this being hand managed for now.